vllm/model_executor/models/aria.py

```diff
@@ -0,0 +1,1084 @@
+import math
+from typing import Iterable, List, Optional, Tuple
+
+import numpy as np
+import torch
+import torch.nn as nn
+from PIL import Image
+from transformers import LlamaConfig
+from transformers.utils import logging
+from vllm.attention import AttentionMetadata
+from vllm.config import CacheConfig, LoRAConfig, VllmConfig
+from vllm.distributed import (
+    get_tensor_model_parallel_rank,
+    get_tensor_model_parallel_world_size,
+    tensor_model_parallel_all_reduce,
+)
+from vllm.inputs import INPUT_REGISTRY, token_inputs
+from vllm.model_executor.layers.fused_moe import fused_moe
+from vllm.model_executor.layers.layernorm import RMSNorm
+from vllm.model_executor.layers.logits_processor import LogitsProcessor
+from vllm.model_executor.layers.quantization.base_config import QuantizationConfig
+from vllm.model_executor.layers.sampler import Sampler, SamplerOutput, SamplingMetadata
+from vllm.model_executor.layers.vocab_parallel_embedding import ParallelLMHead
+from vllm.model_executor.models.interfaces import SupportsMultiModal
+from vllm.model_executor.models.llama import (
+    LlamaAttention,
+    LlamaDecoderLayer,
+    LlamaMLP,
+    LlamaModel,
+    RMSNorm,
+)
+from vllm.model_executor.models.utils import (
+    AutoWeightsLoader,
+    WeightsMapper,
+    make_layers,
+    maybe_prefix,
+    merge_multimodal_embeddings,
+)
+from vllm.model_executor.utils import set_weight_attrs
+from vllm.multimodal import MULTIMODAL_REGISTRY
+from vllm.multimodal.base import MultiModalInputs
+from vllm.multimodal.image import cached_get_image_processor
+from vllm.multimodal.utils import (
+    cached_get_tokenizer,
+    repeat_and_pad_placeholder_tokens,
+)
+from vllm.sequence import IntermediateTensors
+from vllm.utils import is_list_of
+
+logger = logging.get_logger(__name__)
+
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+from torch.nn.init import trunc_normal_
+from transformers.activations import ACT2FN
+from transformers.models.idefics2.configuration_idefics2 import Idefics2VisionConfig
+from vllm.config import QuantizationConfig
+from vllm.model_executor.models.idefics2_vision_model import Idefics2VisionTransformer
+
+
+class AriaVisionConfig(Idefics2VisionConfig):
+    model_type = "aria_vision_model"
+
+
+class IdentityOp(torch.nn.Module):
+    """
+    An identity operation that returns the input unchanged.
+
+    This can be used as a placeholder or to maintain architectural consistency
+    when a specific operation is not needed.
+    """
```

Comment:
Perhaps we can use `nn.Identity` directly? Seems that `IdentityOp` is only used to placehold `Idefics2VisionTransformer`'s post_layernorm.

Reply:
Yes! I didn't even know about the existence of `nn.Identity`; that's exactly what I needed. Thanks!